conda/cli/install.py

```diff
@@ -361,76 +399,65 @@ def install(args, parser, command="install"):
         not in (UpdateModifier.FREEZE_INSTALLED, UpdateModifier.UPDATE_SPECS)
     ) and not newenv
 
-    for repodata_fn in repodata_fns:
+    update_modifier = context.update_modifier
+    if (isinstall or isremove) and args.update_modifier == NULL:
+        update_modifier = UpdateModifier.FREEZE_INSTALLED
+    deps_modifier = context.deps_modifier
+    if isupdate:
+        deps_modifier = context.deps_modifier or DepsModifier.UPDATE_SPECS
+
+    @retry_package_not_found(
+        repodata_fns=repodata_fns,
+        index_args=index_args,
+        retry_extra_errors=(
+            UnsatisfiableError,
+            SpecsConfigurationConflictError,
+            SystemExit,
+        ),
+    )
```

Comment:
Having a decorated inline function is a bit strange. Would it be better to have `retry_package_not_found` as a context manager instead?

Reply:
ya, true! Giving a context manager a go

Reply:
@jaimergp I've updated this part of the code to use a context manager approach. Does this look better to you? Will add some extra unit tests to ensure that its expressing the right behaviour if this is the direction we want to go.